ratis-server/src/test/java/org/apache/ratis/statemachine/SimpleStateMachine4Testing.java

```diff
@@ -358,20 +361,26 @@ public TransactionContext startTransaction(RaftClientRequest request) {
 
   @Override
   public CompletableFuture<Void> write(LogEntryProto entry) {
-    blocking.await(Blocking.Type.WRITE_STATE_MACHINE_DATA);
-    return CompletableFuture.completedFuture(null);
+    return CompletableFuture.supplyAsync(() -> {
+      blocking.await(Blocking.Type.WRITE_STATE_MACHINE_DATA);
+      return null;
+    }, executor);
```

Comment:
I think we can use another api in SimpleStateMachine4Testing#Blocking class which just returns a future which is not yet completed. This api would be called here to return a future for  write(LogEntryProto entry) call. unblockWriteStateMachineData can then complete that future. This would avoid a blocking call.

Reply:
Good point! Done.